### PR TITLE
Remove commented and unneeded code

### PR DESCRIPTION
**Pull request by marbar3778**
- Remove commented out file
- Remove colors
- remove newdebugdb
Currently newdebugDB is used in cosmos-sdk for testing and iavl for testing, If this pr is merged following PRs will be made to eliminate the code in those repos as well
Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: libs/db/debug_db.go [hunk 1]
```diff
@@ -1,257 +0,0 @@
...
-	"fmt"
-	"sync"
-
-	cmn "github.com/tendermint/tendermint/libs/common"
-)
-
-//----------------------------------------
-// debugDB
-
-type debugDB struct {
```
Comment (melekes) [inline libs/db/debug_db.go]: this can be used for debugging. I've never used it myself, but maybe Cosmos-SDK devs had & need it.

Comment (marbar3778) [reply libs/db/debug_db.go]: its set as `const = false` in the code in the sdk. cc @alexanderbez @alessio @jackzampolin @rigelrozanski have you guys used newdbDebug at anytime

**Comment by melekes** [comment]
This needs a changelog entry.

```
That said, breaking changes in the following packages will be documented in the
CHANGELOG even if they don't lead to MINOR version bumps:
...
- libs
  - common
  ...
  - db
```

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ program](https://hackerone.com/tendermint).
 * Apps
 
 * Go API
+  - [libs] Removal of `NewDebugDB`, `colors.go` & `errors.go` (commented out entire file) (@marbar3778)
```
Comment (melekes) [inline CHANGELOG_PENDING.md]:
```suggestion
  - [libs] Remove unused `db/debugDB` and `common/colors.go` & `errors/errors.go` files (@marbar3778)
```

**Pull request merged**